cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -265,7 +277,10 @@ private void computeBestProposal() {
     }
   }
 
-  private boolean validCachedProposal() {
+  private boolean validCachedProposal() throws KafkaCruiseControlException {
```

Comment:
semantically, _executor.hasOngoingExecution() means cache is potentially not valid, should this be in a different function?

Reply:
Checking cache validity during an ongoing execution is a sanity check failure; hence, I feel it should be a part of this validity check.